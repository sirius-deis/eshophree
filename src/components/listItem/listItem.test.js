import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ListItem from './listItem';

describe('ListItem component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <ListItem>test</ListItem>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render component with correct value', () => {
    render(
      <MemoryRouter>
        <ListItem>test</ListItem>
      </MemoryRouter>,
    );
    const component = screen.queryByText(/test/i);
    expect(component).toBeInTheDocument();
    expect(component.textContent).toBe('test');
  });
  it('should render component which contain - with correct value', () => {
    render(
      <MemoryRouter>
        <ListItem>test-test</ListItem>
      </MemoryRouter>,
    );
    const component = screen.queryByText(/test/i);
    expect(component).toBeInTheDocument();
    expect(component.textContent).toBe('test test');
  });
});
