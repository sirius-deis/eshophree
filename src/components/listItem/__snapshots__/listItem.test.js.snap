// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should match snapshot 1`] = `
<div>
  <li
    class="sc-bgqQcB lnKBne"
  >
    <button
      class="sc-gTRrQi iMirOm"
    >
      test
    </button>
  </li>
</div>
`;
